astropy/io/misc/tests/test_hdf5.py

```diff
@@ -872,80 +872,3 @@ def test_hdf5_mixins_per_column(table_cls, name_col, tmpdir):
     if name.startswith('tm'):
         assert t2[name]._time.jd1.__class__ is np.ndarray
         assert t2[name]._time.jd2.__class__ is np.ndarray
-
-
-@pytest.mark.skipif('HAS_YAML or not HAS_H5PY')
-def test_warn_for_dropped_info_attributes(tmpdir):
-    filename = str(tmpdir.join('test.hdf5'))
-    t = Table([[1, 2]])
-    t['col0'].info.description = 'hello'
-    with pytest.warns(AstropyUserWarning, match=r"table contains column\(s\) "
-                      r"with defined 'unit'") as warns:
-        t.write(filename, path='root', serialize_meta=False)
-    assert len(warns) == 1
-
-
-@pytest.mark.skipif('HAS_YAML or not HAS_H5PY')
-def test_error_for_mixins_but_no_yaml(tmpdir):
-    filename = str(tmpdir.join('test.hdf5'))
-    t = Table([mixin_cols['sc']])
-    with pytest.raises(TypeError) as err:
-        t.write(filename, path='root', serialize_meta=True)
-    assert "cannot write type SkyCoord column 'col0' to HDF5 without PyYAML" in str(err.value)
-
-
-@pytest.mark.skipif('not HAS_YAML or not HAS_H5PY')
```

Comment:
Are you sure we can delete this test?

Reply:
We now always have yaml... I think so. I am worried about those couple of tests, but glancing at the code, it seems to need one of yaml  or h5py.

Reply:
I mean, shouldn't we keep this test but with `@pytest.mark.skipif('not HAS_H5PY')`

Reply:
If the tested code needs either yaml or h5py , then I'm not sure the test would ever do anything. IDK. I'm not familiar with this part of the code.

Reply:
I'm also confused. Seems like this would become `skipif('not HAS_H5PY')`

Reply:
I am confused now. You didn't delete tests in `astropy/io/misc/tests/test_hdf5.py` and they had similar skip logic.

Reply:
hmm. I'll reimplement those tests and let relevant maintainers prune them. Path of least resistance.

Reply:
The `skipif` skips if either yaml of hdf5 is missing, so it needs both.